app.rb

```diff
@@ -1,7 +1,47 @@
 require 'sinatra/base'
+require './lib/game'
+require './lib/player'
+require './lib/computer_player'
+require './lib/rules'
+
 class RockPaperScissors < Sinatra::Base
-  get '/test' do
-    'test page'
+
+  get '/' do
+    erb :index
+  end
+
+  post '/name' do
+    player_1 = Player.new(params[:player_1_name])
+    player_2 = ComputerPlayer.new(Rules.new)
+    @game = Game.create(player_1, player_2, Rules.new)
+    redirect'/play'
+  end
+
+  before do
+    @game = Game.instance
+  end
+
+  get '/play' do
+    erb :play
+  end
+
+  post '/make-move' do
+    @game.player_1_move(params[:move].to_sym)
+    redirect '/confirm'
+  end
+
+  get '/confirm' do
+    @move = @game.player1_move
+    erb :confirm
+  end
+
+  post '/computer_turn' do
+    @game.player_2_move
+    redirect '/results'
+  end
+
+  get '/results' do
+    erb @game.result(@game.player1_move, @game.player2_move)
```

Comment:
That's a smart way to use the result symbol as the view name - however, in a real program, we probably want to be careful with this. It's fine here, since we know there won't be any other value than :win, :lose or :draw, but it's always worth double checking (otherwise the controller doesn't really get to decide which view is called, because the view name depends directly on what the model layer (`Game`) returns)